### PR TITLE
Cleanup scripts

### PR DESCRIPTION
**Pull request by TimothyJones**
Looking at #584 highlighted that it's not super clear when things are triggered with our package.json scripts. I've started to clean up and consolidate them to hopefully make it clearer - thoughts welcome.

I like the guideline that the scripts are named after what they're for, not what they invoke. Like `test` instead of `jest`. Under that guideline, there's still a bit of work to do here.

(Matt, I know you don't have a lot of time atm, I'm mostly tagging you in case you're interested)

Things I'm thinking about:

* Do we need all the examples listed in package-json? We don't use those scripts in CI.
* I'm not sure whether the output of jscpd is used
* We're still generating coverage for coveralls, but aren't sending it to them (this is probably an easy fix, if we want to keep publishing those).
* I think we can also remove the `webpack` script

All thoughts and comments welcome.

### CODE REVIEW

**Comment by TimothyJones** [comment]
@individual-it : I wanted to add you as a reviewer, but you didn't come up in the drop down for some reason. Would be keen to know what you think.

**Comment by mefellows** [comment]
> Do we need all the examples listed in package-json? We don't use those scripts in CI.

I don't think so. We just want to be able run the examples as part of our CI process as a form of feature spec (and of course to make sure they are actually runnable and therefore something people can use). To be honest though, we ought to have a more dedicated feature suite - for another time.

> I'm not sure whether the output of jscpd is used

It used to be, it's quite frankly not something I'm too concerned about so I'm happy to see it removed. I think between us we'll be able to judge with PRs etc. if we're badly duplicating code. There is one area of the code base (pact-web) which could be refactored, I don't need jscpd to help me recognise that.

> We're still generating coverage for coveralls, but aren't sending it to them (this is probably an easy fix, if we want to keep publishing those).

I'm not super concerned about sending the coverage reports to coveralls (although I did think they were still working, not that I've looked recently). The main value is having some guidance around testing and %'s just give us a sense of that. Don't need coveralls to do that for us.

> I think we can also remove the webpack script

Webpack was needed by the pact-web distribution. As long as that can still build and release then the script step can go.

**Comment by mefellows** [comment]
BTW the script directory cleanup looks much nicer - thanks heaps for doing it.

**Review by individual-it — CHANGES_REQUESTED** [review]
I cannot see where lint and format are run in CI
I actually would propose to run them directly as first steps in CI, so its very obvious for the developers

File: package.json [hunk 1]
```diff
@@ -5,20 +5,17 @@
...
+    "dist": "tsc && webpack --config ./config/webpack.web.config.js && copyfiles package.json ./dist",
     "jscpd": "jscpd -p src -r json -o jscpd.json",
-    "lint": "npm run lint:prettier:ts && npm run lint:prettier:js && tslint -c tslint.json '{src,test,examples}/**/*.ts' -e '**/node_modules/**/*.ts'",
-    "lint:fix": "prettier --parser typescript --write \"{src,test,examples}/**/*.ts\" && prettier --write \"{src,test,examples}/**/*.js\"",
-    "lint:prettier:ts": "prettier --parser typescript --check \"{src,test,examples}/**/*.ts\"",
-    "lint:prettier:js": "prettier --check \"{src,test,examples}/**/*.js\"",
+    "lint": "tslint -c tslint.json '{src,test,examples}/**/*.ts' -e '**/node_modules/**/*.ts'",
+    "lint:fix": "npm run lint -- --fix",
+    "format:base": "prettier \"{src,test,examples}/**/*.{ts,js}\"",
+    "format:fix": "npm run format:base -- --fix",
```
[the remark this answers is not in the document]
Comment (TimothyJones) [reply package.json:15]: You're right, I'll fix that.

File: package.json [hunk 2]
```diff
@@ -5,20 +5,17 @@
   "main": "./pact.js",
   "types": "./pact.d.ts",
   "scripts": {
-    "build": "npm run lint && npm run compile",
-    "compile": "rimraf dist && tsc",
     "clean": "rimraf docs dist dist-web coverage .nyc_output logs pacts jscpd.json",
     "coverage": "nyc report --reporter=lcov",
-    "deploy:package": "tar -czf pactjs.tar.gz config dist src package.json README.md LICENSE",
-    "deploy:prepare": "npm i --production && npm run deploy:package",
```
[the remark this answers is not in the document]
Comment (TimothyJones) [reply package.json:13]: Ah, well spotted, thanks.

**Pull request merged**